### PR TITLE
Fix issue #174

### PR DESCRIPTION
**Pull request by kungknut**
When merged, this will fix issue #174.

### CODE REVIEW

**Comment by qarizma** [comment]
This is a very annoying bug in my opinion. This should fix it, however it's not efficient because you are doing database operations twice, which could be done in one.
```
$this->deleteExistingSessions($uid);
$this->addSession($uid, false);
```

Let's see if we can make a refreshSession() function for this 👍 
For now, please merge because it's important.

File: Auth.php [hunk 1]
```diff
@@ -462,7 +462,7 @@ public function checkSession($hash)
         $sid = $row['id'];
         $uid = $row['uid'];
         $expiredate = strtotime($row['expiredate']);
-        $currentdate = strtotime(date("Y-m-d H:i:s"));
```
Comment (liamjack) [inline Auth.php]: Wow this is inefficient

**Comment by liamjack** [comment]
I'm not sure that this is the best solution, this recreates a session when an existing session expires. In this case what use is the session expiry date ?

**Comment by qarizma** [comment]
@liamjack you are right, this suggested code isn't the correct way. You should only change the date/time of last access.

The expire feature is for people that aren't browsing the site anymore, then it should expire. But it should NOT expire when I'm browsing the site.

**Comment by kungknut** [comment]
I am of the same opinion as @qarizma. As long as the user is actively browsing the site, the session should not expire. It should expire if the user has been inactive for a certain amount of time, which is the time stored in `cookie_forget` or, if the user ticked "remember me" `cookie_remember`.

The line you mention is inefficient @liamjack, I have suggested a change of that in this PR, do you think it will suffice?

This PR may not present the absolutely best solution to the problem it's stated to fix, but why not go the the way @qarizma proposed above and write the function `refreshSession()` instead, or merge this and add that function to the todo-list?

Maybe this could be a good opportunity to let PHPAuth set the cookie, it is however essential for this system to work. I think it would be semantically correct to include that as a part of the `addSession()`-function.

**Comment by liamjack** [comment]
@kungknut Yeah the change you implemented will definitely improve that inefficient line, I was just impressed because it's probably my code from a few years ago :)

The problem is that the current PR code will recreate any expired session, whether it's been expired for 2 minutes or 2 years. Maybe only recreate the session if it expired within the past 10 minutes, or if not the session expiration could be updated every time the session is validated successfully (maybe not the best solution from a DB optimization standpoint)

**Comment by kungknut** [comment]
@liamjack, are you one of the core authors of this project? This is probably off topic, but out of curiousity, may I ask you why you choose to use the strtotime compatible time formats and what the advantage is (or was) compared to other formats?

Have a look at line 470, there is an if statement that check if the session is still valid, if not the function will return false. That in turn means the session won't refresh if it has already expired, or am I missing something?

**Comment by liamjack** [comment]
@kungknut Yes I'm the original author of PHPAuth, I've had to slow down contributions due to a lack of time. There's no valid reason for using `strtotime` all over the place, it's just how I did things at the time. It's pretty inefficient, we could just store the timestamp instead of the date representation and stop all the useless date -> timestamp conversions.

Indeed you're right about line 470, I read too quickly. So if I understand correctly this will create a new session for sessions that only have `[$cookie_renew]` time left ? (You're going to have to define a default value in the SQL dumps btw)

**Comment by kungknut** [comment]
@liamjack, I understand, sometimes life just happens and the time disappear :). I agree about the timestamp, I'll see if I can manage to make a PR regarding this.

That is correct about `cookie_renew`. That means, with the default settings, the session will be renewed every 25minutes, if the user is actively browsing. Maybe the fact that it's done inefficiently using two database query's in a row does not impact performance that much thanks to the time delay. There is a default setting added in this PR already.

What do you think about adding an option to the `addSession()` to choose whether PHPAuth should set the cookie? (Proposed above)

File: Auth.php (hunk 1 above)
Comment (Conver) [reply Auth.php]: Whats your take on this @liamjack?

**Pull request merged**